### PR TITLE
Rwc2018/challenge gpsr2

### PR DESCRIPTION
**Pull request by MatthijsBurgh**

### CODE REVIEW

File: challenge_gpsr/src/gpsr.py [hunk 1]
```diff
@@ -19,6 +19,196 @@
...
+                                                                              id=self.knowledge.exit_waypoint),
+                                     radius=0.3)
+            self.robot.speech.speak("I'm done now. Thank you very much, and goodbye!", block=True)
+            nwc.execute()
+            return
+
+    def _on_request_missing_information(self, description, grammar, target):
+        rospy.loginfo("_request_missing_information('{}', '{}...', '{}')".format(description, grammar[:10], target))
+
+        while True:
```
Comment (MatthijsBurgh) [inline challenge_gpsr/src/gpsr.py]:
```suggestion
        while not rospy.is_shutdown():
```

File: challenge_gpsr/src/gpsr.py [hunk 2]
```diff
@@ -19,6 +19,196 @@
...
+        self.robot.rightArm.send_gripper_goal('close',0.0)
+        self.robot.torso.reset()
+
+        self.robot.speech.speak("Trigger me by saying my name, and wait for the ping.", block=True)
+
+        self.wait_to_be_called()
+
+        self.robot.speech.speak("What can I do for you?", block=True)
+
+        while True:
```
Comment (MatthijsBurgh) [inline challenge_gpsr/src/gpsr.py]:
```suggestion
        while not rospy.is_shutdown():
```

File: challenge_gpsr/src/gpsr.py [hunk 3]
```diff
@@ -19,6 +19,196 @@
 
 from robocup_knowledge import load_knowledge
 
+from conversation_engine import ConversationEngine
+
+class ConversationEngineWithHmi(ConversationEngine):
```
Comment (MatthijsBurgh) [inline challenge_gpsr/src/gpsr.py]: @LoyVanBeek This should be moved to conversation_engine.

Comment (MatthijsBurgh) [reply challenge_gpsr/src/gpsr.py]: @LoyVanBeek ping ;)

Comment (LoyVanBeek) [reply challenge_gpsr/src/gpsr.py]: This is the core of the challenge, I'd rather keep this right here.

Comment (MatthijsBurgh) [reply challenge_gpsr/src/gpsr.py]: also not move it to a separate file in this package?

**Pull request merged**